pkg/northbound/gnmi/v2/get.go

```diff
@@ -171,21 +171,34 @@ func (s *Server) processRequest(ctx context.Context, req *gnmi.GetRequest, group
 				ch := make(chan configapi.ConfigurationEvent)
 				err := s.configurations.Watch(ctx, ch, configuration.WithConfigurationID(configuration.NewID(target.targetID)), configuration.WithReplay())
 				if err != nil {
-					return nil, err
+					return nil, errors.Status(err).Err()
 				}
 				wg.Add(1)
-				go func(id configapi.ConfigurationID) {
+				go func(target *targetInfo) {
+					defer wg.Done()
 					for event := range ch {
 						if event.Configuration.Status.Applied.Index >= target.configuration.Status.Committed.Index &&
 							event.Configuration.Status.Applied.Term == event.Configuration.Status.Term {
-							wg.Done()
 							return
 						}
 					}
-				}(target.configuration.ID)
+				}(target)
```

Comment:
Because of the way Go's `for` loops work, we have to pass the target info into the goroutine, otherwise the `target.configuration.Status.Committed.Index` used in the condition above will basically just use the last target in the loop. I doubt this would have worked with multiple targets.

Reply:
yep good catch.